pkg/controllers/plugin/plugin_workload_status_controller.go

```diff
@@ -0,0 +1,325 @@
+// If the Plugin has a clusterName set, the RestClientGetter is initialized from the cluster secret.
+// Otherwise, the RestClientGetter is initialized with in-cluster config
+func (r *WorkLoadStatusReconciler) initClientGetter(
+	ctx context.Context,
+	plugin greenhousev1alpha1.Plugin,
+	pluginStatus greenhousev1alpha1.PluginStatus,
+) (
+	clusterAccessReadyCondition greenhousev1alpha1.Condition,
+	restClientGetter genericclioptions.RESTClientGetter,
+) {
+
+	clusterAccessReadyCondition = *pluginStatus.GetConditionByType(greenhousev1alpha1.ClusterAccessReadyCondition)
+	clusterAccessReadyCondition.Status = metav1.ConditionTrue
+
+	var err error
+
+	// early return if spec.clusterName is not set
+	if plugin.Spec.ClusterName == "" {
+		restClientGetter, err = clientutil.NewRestClientGetterForInCluster(plugin.GetReleaseNamespace(), r.kubeClientOpts...)
+		if err != nil {
+			clusterAccessReadyCondition.Status = metav1.ConditionFalse
+			clusterAccessReadyCondition.Message = fmt.Sprintf("cannot access greenhouse cluster: %s", err.Error())
+			return clusterAccessReadyCondition, nil
+		}
+		return clusterAccessReadyCondition, restClientGetter
+	}
+
+	cluster := new(greenhousev1alpha1.Cluster)
+	err = r.Get(ctx, types.NamespacedName{Namespace: plugin.Namespace, Name: plugin.Spec.ClusterName}, cluster)
+	if err != nil {
+		clusterAccessReadyCondition.Status = metav1.ConditionFalse
+		clusterAccessReadyCondition.Message = fmt.Sprintf("Failed to get cluster %s: %s", plugin.Spec.ClusterName, err.Error())
+		return clusterAccessReadyCondition, nil
+	}
+
+	readyConditionInCluster := cluster.Status.StatusConditions.GetConditionByType(greenhousev1alpha1.ReadyCondition)
+	if readyConditionInCluster == nil || readyConditionInCluster.Status != metav1.ConditionTrue {
+		clusterAccessReadyCondition.Status = metav1.ConditionFalse
+		clusterAccessReadyCondition.Message = fmt.Sprintf("cluster %s is not ready", plugin.Spec.ClusterName)
+		return clusterAccessReadyCondition, nil
+	}
+
+	// get restclientGetter from cluster if clusterName is set
+	secret := corev1.Secret{}
+	err = r.Get(ctx, types.NamespacedName{Namespace: plugin.Namespace, Name: plugin.Spec.ClusterName}, &secret)
+	if err != nil {
+		clusterAccessReadyCondition.Status = metav1.ConditionFalse
+		clusterAccessReadyCondition.Message = fmt.Sprintf("Failed to get secret for cluster %s: %s", plugin.Spec.ClusterName, err.Error())
+		return clusterAccessReadyCondition, nil
+	}
+	restClientGetter, err = clientutil.NewRestClientGetterFromSecret(&secret, plugin.GetReleaseNamespace(), r.kubeClientOpts...)
+	if err != nil {
+		clusterAccessReadyCondition.Status = metav1.ConditionFalse
+		clusterAccessReadyCondition.Message = fmt.Sprintf("cannot access cluster %s: %s", plugin.Spec.ClusterName, err.Error())
+		return clusterAccessReadyCondition, nil
+	}
+	clusterAccessReadyCondition.Status = metav1.ConditionTrue
+	clusterAccessReadyCondition.Message = ""
+	return clusterAccessReadyCondition, restClientGetter
+}
```

Comment:
This is copy+paste from https://github.com/cloudoperators/greenhouse/blob/ca40292434c3ad1cff718d5dbfe86a648fb1d0db/pkg/controllers/plugin/helm_controller.go#L170-L227

Since we need this now in more than one place, could you refactor it into a package private method. I think we should limit it to the package since this is very plugin specific.

Reply:
@IvoGoman Good point! `initClientGetter ` will be also required for this [controller](https://github.com/cloudoperators/greenhouse/issues/106) to do periodic Helm Chart Test. @kengou Pull me in anytime if you want to do some pair programming :)

Reply:
@ibakshay thanks will do definitely 😉

Reply:
done [e4f0f4f](https://github.com/cloudoperators/greenhouse/pull/398/commits/e4f0f4f6868c2425b7214270c689b9d5146c8f4e)